src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -302,6 +302,31 @@ private void evaluateSystemIndicesAccess(
                 }
         }
 
+        if (user.isPluginUser()) {
+            Set<String> matchingSystemIndices = SystemIndexRegistry.matchesPluginSystemIndexPattern(
+                user.getName().replace("plugin:", ""),
+                requestedResolved.getAllIndices()
+            );
+            if (requestedResolved.getAllIndices().equals(matchingSystemIndices)) {
+                // plugin is authorized to perform any actions on its own registered system indices
```

Comment:
This code is called for index and cluster actions. Even though `getAllIndices()` probably returns `*` for cluster action as a more or less suitable fallback - shouldn't we better check the action type?

Reply:
Would it be sufficient to change the top level condition to `if (user.isPluginUser() && !requestedResolved.isLocalAll()) { ... }`